server/visualizer/handler.go

```diff
@@ -0,0 +1,260 @@
+package visualizer
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/eukarya-inc/reearth-plateauview/server/cms"
+	"github.com/labstack/echo/v4"
+	"github.com/samber/lo"
+)
+
+type Handler struct {
+	DataModelKey                 string
+	TemplateModelKey             string
+	DataModelDataFieldID         string
+	DataModelIDFieldID           string
+	TemplateModelTemplateFieldID string
+	TemplateModelIDFieldID       string
+	CMS                          cms.Interface
+}
+
+func NewHandler(CMS cms.Interface, dKey, tKey string) (*Handler, error) {
+	if dKey == "" || tKey == "" {
+		return nil, fmt.Errorf("missing model keys, dataKey=%s, templateKey=%s", dKey, tKey)
+	}
+
+	ctx := context.Background()
+	data, err := CMS.GetItems(ctx, dKey)
+	if err != nil {
+		return nil, err
+	}
+
+	templates, err := CMS.GetItems(ctx, tKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(data.Items) == 0 || len(templates.Items) == 0 {
+		return nil, fmt.Errorf("failed to fetch meta data")
+	}
+
+	h := &Handler{
+		DataModelKey:                 dKey,
+		TemplateModelKey:             tKey,
+		CMS:                          CMS,
+		DataModelIDFieldID:           data.Items[0].Fields[0].ID,
+		DataModelDataFieldID:         data.Items[0].Fields[1].ID,
+		TemplateModelIDFieldID:       templates.Items[0].Fields[0].ID,
+		TemplateModelTemplateFieldID: templates.Items[0].Fields[1].ID,
+	}
+
+	return h, nil
+}
+
+// GET | /viz/:id
+func (h *Handler) fetchRoot() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		data, err := h.CMS.GetItems(ctx, h.DataModelKey)
```

Comment:
```suggestion
		data, err := h.CMS.GetItemsByKey(ctx, h.CMSProject, h.DataModelKey)
```

Also an proejct ID is requried to get items by a model key.

the same applies hereafter